File: ViLabWebClient/src/app/models/admin/read/subcategory.ts
export class Subcategory {
  public id: number;
  public name: string;
  public categoryId: number;

  constructor(model) {
    this.id = model == null ? 0 : model.id;
    this.name = model == null ? "" : model.name;
    this.categoryId = model == null ? "" : model.categoryId;
  }
}

File: DbModel/Entities/Subcategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace DbModel.Entities
{
    public class Subcategory
  {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string Description { get; set; }

        [ForeignKey("CategoryId")]
        public Category Category { get; set; }

        public int CategoryId { get; set; }

        public ICollection<Case> Cases { get; set; } = new List<Case>();
    }
}
